### PR TITLE
Commons: Update SufficientBalanceValidator.validate implementation

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-commons/src/core/utils/isDataForFunctionCall.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import {utils} from 'ethers';
+
+export const isDataForFunctionCall = (data: string, contractInterface: utils.Interface, functionName: string) => {
+  const functionSignature = contractInterface.functions[functionName].sighash;
+  return functionSignature === data.slice(0, functionSignature.length);
```
Comment (Marik-D) [inline universal-login-commons/src/core/utils/isDataForFunctionCall.ts]:
```suggestion
  return data.startsWith(functionSignature)
```

File: universal-login-relayer/src/core/utils/encodeData.ts [hunk 2]
```diff
@@ -1,16 +1,16 @@
 import {utils} from 'ethers';
-import {beta2} from '@universal-login/contracts';
 import {ensure} from '@universal-login/commons';
+import {beta2} from '@universal-login/contracts';
 import {InvalidHexData} from './errors';
 
 export const isDataForFunctionCall = (data: string, contract: any, functionName: string) => {
```
Comment (Marik-D) [inline universal-login-relayer/src/core/utils/encodeData.ts]: duplicate

File: universal-login-commons/test/core/utils/getFeeCurrencyValueFrom.unit.test.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import {utils} from 'ethers';
+import {TEST_PRIVATE_KEY, ETHER_NATIVE_TOKEN} from '../../../src';
+import {getTestSignedMessage} from '../../helpers/getTestMessage';
+import {getFeeCurrencyValueFrom} from '../../../src/core/utils/getFeeCurrencyValueFrom';
+
+describe('UNIT: getFeeCurrencyValueFrom', () => {
+  describe('ETH', () => {
+    it('return balance eq 0 when gasPrice is 0', () => {
+      const signedMessage = getTestSignedMessage(TEST_PRIVATE_KEY, {gasPrice: 0, safeTxGas: 100, baseGas: 1000, gasToken: ETHER_NATIVE_TOKEN.address});
+      expect(getFeeCurrencyValueFrom(signedMessage)).to.be.deep.eq({
```
Comment (JustynaBroniszewska) [inline universal-login-commons/test/core/utils/getFeeCurrencyValueFrom.unit.test.ts]: `to.deep.eq`

File: universal-login-commons/src/integration/ethereum/validators/SufficientBalanceValidator.ts [hunk 4]
```diff
@@ -15,11 +16,9 @@ export class SufficientBalanceValidator implements IMessageValidator {
...
   async validate(signedMessage: SignedMessage) {
-    const paymentOptions: Omit<PaymentOptions, 'refundReceiver'> = {
-      gasToken: signedMessage.gasToken,
-      gasLimit: utils.bigNumberify(signedMessage.safeTxGas).add(signedMessage.baseGas),
-      gasPrice: signedMessage.gasPrice,
-    };
-    ensure(await hasEnoughToken(paymentOptions, signedMessage.from, this.balanceChecker), NotEnoughTokens);
+    const feeCurrencyValue = getFeeCurrencyValueFrom(signedMessage);
+    const balance = await this.balanceChecker.getBalance(signedMessage.from, feeCurrencyValue.address);
+    const hasEnoughToken = balance.gte(feeCurrencyValue.balance);
```
Comment (JustynaBroniszewska) [inline universal-login-commons/src/integration/ethereum/validators/SufficientBalanceValidator.ts]: We don't use `hasEnoughToken`, right? Maybe we should delete it.

**Pull request merged**